agent/command_test.go

```diff
@@ -168,3 +195,56 @@ func TestEndTaskSyncCommands(t *testing.T) {
 		})
 	}
 }
+
+func (s *CommandSuite) TestVarsAreUnsetAfterRunning() {
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
nit: since this is a suite, it's generally preferred to just absorb ctx into `SetUpTest` and initialize suite fields with the ctx/cancel func, then cancel the context in `TearDownTest`.